File: test-fixtures/cross-language-typeparams/csharp/Generics.cs
namespace CrossLanguage.Generics;

public interface IA {}
public interface IB {}

public class CsBox<T> where T : IA {
    public T Value;
}

public class Generics {
    public static T Identity<T>(T value) where T : class, new() {
        return value;
    }

    public static T Combine<T>(T value) where T : IA, IB, notnull {
        return value;
    }
}

File: test-fixtures/cross-language-typeparams/kotlin/Generics.kt
package crosslanguage.generics

interface A
interface B

class KBox<out T>(val value: T)

inline fun <reified T> ktIdentity(value: T): T = value

fun <T> ktConstrained(value: T): T where T : A, T : B = value

class KStore<T> where T : A {
    fun <U : B> put(value: U): U = value
}
